### PR TITLE
fix: missing Hubpost contacts throwing errors on download page

### PR DESCRIPTION
**Pull request by carlmw**
Fixes #

Users not being able to download when signing up with an email that was not already a Hubspot contact.

## How to test

1. Go to https://deploy-preview-2906--oak-web-application.netlify.thenational.academy
2. Click on \_\_\_\_\_\_
3. You should see \_\_\_\_\_\_

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/TeacherComponents/helpers/downloadAndShareHelpers/fetchHubspotContactDetails.ts [hunk 1]
```diff
@@ -21,16 +24,24 @@ export async function fetchHubspotContactDetails(email: string) {
...
-    });
+    const error =
+      err instanceof OakError
+        ? err
+        : new OakError({
+            code: "hubspot/unknown",
+            originalError: err,
+          });
+
+    reportError(error);
```
Comment (carlmw) [inline src/components/TeacherComponents/helpers/downloadAndShareHelpers/fetchHubspotContactDetails.ts:43]: We weren't properly reporting this error which made it unhelpful in Bugsnag. This should now populate the `originalError` in Bugsnag metadata which will make it easier to trace in future.

File: src/components/TeacherComponents/helpers/downloadAndShareHelpers/fetchHubspotContactDetails.ts [hunk 2]
```diff
@@ -21,16 +24,24 @@ export async function fetchHubspotContactDetails(email: string) {
       throw new Error("Failed to fetch contact details");
     }
 
+    if (response.status === 204) {
```
Comment (carlmw) [inline src/components/TeacherComponents/helpers/downloadAndShareHelpers/fetchHubspotContactDetails.ts:27]: This was the root. Our endpoint was already set up to handle missing contacts by responding with a 204, but we weren't handling it at all.

**Review by k-huggs — APPROVED** [review]

**Review by jack-skerrett-oak — APPROVED** [review]
There is still a brief moment where the button is greyed out when you load the download page after logging in as a completely fresh user, but it is an improvement and I don't want to delay anything.

**Pull request merged**